[TensorExpr] Add a skeleton for a registry of NNC lowerings.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* #65843
* #65554
* #65553
* #65552
* #65551
* __->__ #65550
* #65549

This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

Commit:
[TensorExpr] Add a skeleton for a registry of NNC lowerings.

This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2455,6 +2455,13 @@ c10::optional<ScalarType> findDtypeForValue(const torch::jit::Value* v) {
 
+NNCLoweringFunction getStandardLoweringFor(const std::string& op) {
+  const auto& lowerings = getNNCLoweringRegistry();
+  if (lowerings.count(op))
```
nit: add braces '{...}', just to keep the coding style consistent.

Commit:
rebase on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

Commit:
Update on "[TensorExpr] Add a skeleton for a registry of NNC lowerings."


This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).
